hpy/tools/autogen/public_api.h

```diff
@@ -1,322 +1,454 @@
-typedef int HPy;
-typedef int HPyContext;
-typedef int HPyModuleDef;
-typedef int HPyType_Spec;
-typedef int HPyType_SpecParam;
-typedef int HPyCFunction;
-typedef int HPy_ssize_t;
-typedef int HPy_hash_t;
-typedef int wchar_t;
-typedef int size_t;
-typedef int HPyFunc_Signature;
-typedef int cpy_PyObject;
-typedef int HPyField;
-typedef int HPyGlobal;
-typedef int HPyListBuilder;
-typedef int HPyTupleBuilder;
-typedef int HPyTracker;
-typedef int HPy_RichCmpOp;
-typedef int HPy_buffer;
-typedef int HPyFunc_visitproc;
-typedef int HPy_UCS4;
-typedef int HPyThreadState;
-// ^ these are not real typedefs: they are there only to make pycparser happy
-// and this comment is not on the top of the file, so that it is not picked up
-// by autodoc
-
 /* HPy public API */
 
+/*
+ * IMPORTANT: In order to ensure backwards compatibility of HPyContext, it is
+ * necessary to define the order of the context members. To do so, use macros
+ * 'HPyAPI_HANDLE(idx)' for context handles and 'HPyAPI_FUNC(idx)' for
+ * context functions. When adding members, it doesn't matter where they are
+ * located in this file. It's just important that the maximum context index is
+ * incremented by exactly one.
+ */
+
```

Comment:
`HPyAPI_FUNC` is already defined in `hpy.h` though, so it's a bit confusing to use the same name here.
I'll write some suggestions for alternatives below.